### ISSUE TITLE
Savegame won't load with anything newer than 6.4.0.0

### ISSUE THREAD

**Issue by zinni97** [Courseplay/courseplay#7324]
Hey guys

Not a huge important issue here, I'm just reporting it to let you guys know. 

So I've been playing on the ModHub release CP6.4.0.0 pretty much since it came. Today I checked in here and saw the new V6.4.0.3 so I upgraded. 
When I tried to load the savegame with CP6.4.0.3 the game got stuck in a loop while loading. When I downgrade to 6.4.0.0 again it loads fine. 
Other savegames work fine with 6.4.0.3

I also tried loading and saving the savegame without CP, basically "deactivating" CP and then loading with 6.4.0.3 upon that, same result. 

I've tired loading that savegame with CP6.4.0.1 gives the same result and log as 6.4.0.3. 
So in short that savegame only loads with 6.4.0.0. Again other savegames work fine. 

Now just to be clear I don't expect a fix, it seems to be a quite specific problem. 
Obviously if you can think of something I can try I happily will. 

Thanks for your great work guys!

[log.txt](https://github.com/Courseplay/courseplay/files/6750589/log.txt)

**Comment by Tensuko** [comment]
2021-07-01 21:51 Error: Running LUA method 'loadSharedI3DFileFinished'.
2021-07-01 21:51 dataS/scripts/vehicles/specializations/WorkArea.lua:356: attempt to index field 'workAreas' (a nil value)

**Comment by Tensuko** [comment]
@schwiti6190 
We had that Error already in another Issue.
Do you know what have changed about WorkArea in 6.4.0.X ?

**Comment by Commissar0617** [comment]
i may be completely wrong, but i did a bit of digging, the only place that method is used is in toolmanager.lua

near as I can tell, the only change to that file between 6.4.0 and 6.4.1 was c1ae55d93bfbe8ce3e022b71ce9cfa5f4513e2f6

courseplay:setCpMode(vehicle, courseplay.MODE_TRANSPORT)

changed to 

courseplay:setCpMode(vehicle, courseplay.MODE_DEFAULT)


but idk how tf that could cause the nil return tho. maybe im just spitballing. or perhaps GIANTS did something, but last game update was mar 16th according to steam so idk.

**Comment by Commissar0617** [comment]
any progress with this? it is still occurring with 6.4.4. LMK if you want logs or testing

**Comment by Commissar0617** [comment]
hmmm... i may be a conflict with teh syntrac. my game now locks up when trying to purchase it
